api/envoy/extensions/filters/http/thrift_to_metadata/v3/thrift_to_metadata.proto

```diff
@@ -0,0 +1,137 @@
+option java_outer_classname = "ThriftToMetadataProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/thrift_to_metadata/v3;thrift_to_metadatav3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Thrift-To-Metadata Filter]
+//
+// The Thrift to Metadata filter serves for thrift over HTTP traffic, expecting serialized
+// Thrift request and response bodies in the HTTP payload. It extracts thrift metadata from the
+// HTTP body and put them into the filter metadata. This is useful for matching load balancer
+// subsets, logging, etc.
+//
+// Thrift to Metadata :ref:`configuration overview <config_http_filters_thrift_to_metadata>`.
+// [#extension: envoy.filters.http.thrift_to_metadata]
+
+enum Field {
+  // The Thrift method name, string value.
+  METHOD_NAME = 0;
+
+  // The Thrift protocol name, string value. Values are "binary", "binary/non-strict", and "compact".
+  PROTOCOL = 1;
+
+  // The Thrift transport name, string value. Values are "framed", "header", and "unframed".
+  TRANSPORT = 2;
+
+  // The Thrift message type, singed 16-bit integer value.
+  HEADER_FLAGS = 3;
+
+  // The Thrift sequence ID, singed 32-bit integer value.
+  SEQUENCE_ID = 4;
+
+  // The Thrift message type, string value. Values in request are "call" and "oneway", and in response are "reply" and "exception".
+  MESSAGE_TYPE = 5;
+
+  // The Thrift reply type, string value. This is only valid for response rules. Values are "success" and "error".
+  REPLY_TYPE = 6;
+}
+
+message KeyValuePair {
+  // The namespace — if this is empty, the filter's namespace will be used.
+  string metadata_namespace = 1;
+
+  // The key to use within the namespace.
+  string key = 2 [(validate.rules).string = {min_len: 1}];
+
+  // When used for on_present case, if value is non-empty it'll be used instead
+  // of the field value.
+  //
+  // When used for on_missing case, a non-empty value must be provided.
+  google.protobuf.Value value = 3;
+}
+
+message FieldSelector {
+  option (xds.annotations.v3.message_status).work_in_progress = true;
+
+  // field name to log
+  string name = 1 [(validate.rules).string = {min_len: 1}];
+
+  // field id to match
+  int32 id = 2 [(validate.rules).int32 = {lte: 32767 gte: -32768}];
+
+  // next node of the field selector
+  FieldSelector child = 3;
+}
+
+message Rule {
+  // The field to match on. If set, takes precedence over field_selector.
+  Field field = 1;
+
+  // Specifies that a match will be performed on the value of a field. If set,
+  // the whole http body will be buffered to extract the field value, which
+  // may have performance implications.
+  FieldSelector field_selector = 2 [(xds.annotations.v3.field_status).work_in_progress = true];
+
+  // The key-value pair to set in the metadata if the field is present.
+  KeyValuePair on_present = 3;
```

Comment:
Confusingly, when looking at header_to_metadatea.proto, this actually uses the `value` to override, despite the documentation of the `value` field. Did you notice that discrepancy? What is the plan in this filter?

Reply:
We plan to do the same with `header_to_metadata` so I'll probably update the comment.

If the `value` is present in `on_present` and field is also present (e.g., we have method_name in thrift metadata), we put on_present.value.

If the `value` is not present in `on_present`, we put method_name from thrift metadata for example